FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -59,118 +59,90 @@ final class MessageService {
-            switch processedMessage.messageType {
-            case .error(let errorType):
-                switch errorType {
-                case .needPassphrase:
-                    reject(MessageServiceError.missedPassPhrase(rawMimeData))
-                case .keyMismatch:
-                    reject(MessageServiceError.keyMismatch(rawMimeData))
-                default:
-                    reject(MessageServiceError.unknown)
-                }
-            case .plain, .encrypted:
-                resolve(processedMessage)
+        let processedMessage = try await self.processMessage(rawMimeData: rawMimeData, with: decrypted, keys: keys)
+        switch processedMessage.messageType {
+        case .error(let errorType):
+            switch errorType {
+            case .needPassphrase:
+                throw MessageServiceError.missingPassPhrase(rawMimeData)
+            case .keyMismatch:
+                throw MessageServiceError.keyMismatch(rawMimeData)
+            default:
+                throw MessageServiceError.unknown
             }
+        case .plain, .encrypted:
+            return processedMessage
         }
     }
 
-    private func processMessage(rawMimeData: Data, with decrypted: CoreRes.ParseDecryptMsg, keys: [PrvKeyInfo]) throws -> ProcessedMessage {
+    private func processMessage(rawMimeData: Data, with decrypted: CoreRes.ParseDecryptMsg, keys: [PrvKeyInfo]) async throws -> ProcessedMessage {
         let decryptErrBlocks = decrypted.blocks
             .filter { $0.decryptErr != nil }
-
-        let attachments = try decrypted.blocks
+        let attachmentBlocks = decrypted.blocks
             .filter(\.isAttachmentBlock)
-            .compactMap { block -> MessageAttachment? in
-                guard let attMeta = block.attMeta else { return nil }
-                var name = attMeta.name
-                let size = attMeta.length
-                var data = attMeta.data
-
-                if block.type == .encryptedAtt {
-                    data = (try core.decryptFile(encrypted: data, keys: keys, msgPwd: nil).content)
-                    name = name.deletingPathExtension
-                }
-
-                return MessageAttachment(name: name, size: size, data: data)
+        var attachments: [MessageAttachment] = []
+        for block in attachmentBlocks {
+            guard let meta = block.attMeta else { continue }
+            var name = meta.name
+            var data = meta.data
+            var size = meta.length
+            if block.type == .encryptedAtt { // decrypt
+                let decrypted = try await core.decryptFile(encrypted: data, keys: keys, msgPwd: nil)
+                data = decrypted.content
+                name = decrypted.name
+                size = decrypted.content.count
             }
-
+            attachments.append(MessageAttachment(name: name, size: size, data: data))
```

Comment:
I think it'll be better to move this attachment processing to separate method and simplify code a bit, so we don't need to reassign attachment values:
```swift
var attachments: [MessageAttachment] = []
        for block in attachmentBlocks {
            guard let attachment = try await processAttachment(block: block, keys: keys) else { continue }
            attachments.append(attachment)
        }
....
private func processAttachment(block: MsgBlock, keys: [PrvKeyInfo]) async throws -> MessageAttachment? {
        guard let meta = block.attMeta else { return nil }

        if block.type == .encryptedAtt { // decrypt
            let decrypted = try await core.decryptFile(encrypted: meta.data, keys: keys, msgPwd: nil)
            return MessageAttachment(name: decrypted.name,
                                     size: decrypted.content.count,
                                     data: decrypted.content)
        } else {
            return MessageAttachment(name: meta.name, size: meta.length, data: meta.data)
        }
    }
```